etcdserver/v3_server.go

```diff
@@ -464,6 +466,14 @@ func (s *EtcdServer) Authenticate(ctx context.Context, r *pb.AuthenticateRequest
 }
 
 func (s *EtcdServer) UserAdd(ctx context.Context, r *pb.AuthUserAddRequest) (*pb.AuthUserAddResponse, error) {
+	if r.Options == nil || !r.Options.NoPassword {
+		hashedPassword, err := bcrypt.GenerateFromPassword([]byte(r.Password), s.authStore.BcryptCost())
+		if err != nil {
+			return nil, err
+		}
+		r.Password = base64.StdEncoding.EncodeToString(hashedPassword)
```

Comment:
My concern is recycling `Password` field of the request. It will be treated as different things in the API layer and the state machine layer. Should I add a new field for the hashed password to `AuthUserAddRequest`? I'd like to have other people's opinion.

Reply:
i prefer to add a new field for the hashed password to AuthUserAddRequest due to readability. UserChangePassword and UserAdd  in Store/EtcdServer have duplicated code, could we wrap into a function?

Reply:
Thanks, yeah that makes sense. I'll update this PR tomorrow based on your feedback.

Reply:
I added a new field `HashedPassword` to the requests. Also kept `UserAdd()` in authStore and EtcdServer because they are actually not so similar I think. Could you take a look? Thanks!

Reply:
lgtm.thanks.